etna/analysis/utils.py

```diff
@@ -31,8 +33,15 @@ def _prepare_axes(
     return fig, ax
 
 
-def _get_borders_ts(ts: "TSDataset", start: Optional[str], end: Optional[str]) -> Tuple[str, str]:
+def _get_borders_ts(
+    ts: "TSDataset", start: Optional[Union["TimestampType", str]], end: Optional[Union["TimestampType", str]]
+) -> Tuple[str, str]:
     """Get start and end parameters according to given TSDataset."""
+    from etna.datasets.utils import _check_timestamp_param
```

Comment:
Why do we import this function inside a body?

Reply:
As you can see higher in this file we import entities from `etna.datasets` only under `TYPE_CHECKING`. It is done because of problems with cycle imports. So, any non-type checking imports should be done inside the functions.